providers/aws/services/autoscaling/autoscaling_find_secrets_ec2_launch_configuration/autoscaling_find_secrets_ec2_launch_configuration.py

```diff
@@ -0,0 +1,46 @@
+import os
+from base64 import b64decode
+
+from detect_secrets import SecretsCollection
+from detect_secrets.settings import default_settings
+
+from lib.check.models import Check, Check_Report
+from providers.aws.services.autoscaling.autoscaling_client import autoscaling_client
+
+
+class autoscaling_find_secrets_ec2_launch_configuration(Check):
+    def execute(self):
+        findings = []
+        for configuration in autoscaling_client.launch_configurations:
+            report = Check_Report(self.metadata)
+            report.region = configuration.region
+            report.resource_id = configuration.name
+            report.resource_arn = configuration.arn
+
+            if configuration.user_data:
+                temp_user_data = "/tmp/user_data.txt"
```

Comment:
I think we need to create some temporary file with a random name to prevent different processes writing into it.

Reply:
Also name this variable with something that identifies it as a file handler, like `temp_user_data_file`

Reply:
Done by a machine 🔝